lamindb_setup/dev/upath.py

```diff
@@ -402,6 +396,30 @@ def inner(dir_path: Union[Path, UPath], prefix: str = "", level: int = -1):
 """
 
 
+def retry_access_aws():
+    """Decorator to retry AWS S3 access in case of exception."""
+
+    def wrap_with_args(func):
+        # https://stackoverflow.com/questions/1782843/python-decorator-handling-docstrings
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception as exc:
+                if isinstance(args[0], S3Path):
+                    path = UPath(args[0], cache_regions=True)
```

Comment:
It should always set cache_regions for s3, not only on exception. I would just keep the `create_path` function for this.

Reply:
This is important because `create_path` will also allow all other connections we create with it further (in `lamindb` etc) also have it and prevent problems arising from not having this setting.